### ISSUE TITLE
SPDX headers?

### ISSUE THREAD

**Issue by rossabaker** [typelevel/sbt-typelevel#72]
I have swallowed a bunch of defaults I don't like in the interest of comity, but https://github.com/typelevel/jawn/pull/402 has me wondering about whether this is our opportunity to wave the flag for [SPDX](https://github.com/sbt/sbt-header#using-the-short-spdx-license-identifier-syntax).  It accomplishes the same, is more terse, and is machine readable.  SPDX is good.

**Comment by rossabaker** [comment]
We should crap or get off the pot on this one before we migrate too many more projects.

**Comment by armanbilge** [comment]
I'll merge a PR that makes this change 👍 

With that said, once we have a `prePR` command and we teach Scala Steward to run it for us, we should be able to make changes here and easily propagate them across Typelevel. See also:
- https://github.com/scala-steward-org/scala-steward/pull/2308

**Comment by rossabaker** [comment]
I'm less worried about being able to do it and more worried about defaults churn now that this has been rolled out to more than a couple obscure repositories.

**Comment by armanbilge** [comment]
Fair enough. But it would already be a sort of defaults churn since for most/all repos the de-facto default seems to be non-SPDX headers.

**Comment by rossabaker** [comment]
Yeah, that's a fair point.  I was focued on the repos that have no headers.

There should be a high bar to changing defaults, and this appears to be one that only I care about.

**Comment by DavidGregory084** [comment]
I had a quick look at this one and it seems like it could be awkward as I don't think sbt-header will regenerate existing headers into the new style in `headerCreate`, but it will complain about them using the wrong style for `headerCheck`.

All I did to test this was to add the `headerLicenseStyle` to the `projectSettings` of `TypelevelPlugin` in sbt-typelevel, so it might be that more configuration was needed.

**Comment by rossabaker** [comment]
This is when http4s migrated from SPDX to the default: https://github.com/Banno/mobile-data-services/pull/8954.  I wish I remembered whether that regeneration commit just worked or if I did black sed magic to strip the original header before recreating. 🤔
